src/device.hpp

```diff
@@ -300,10 +300,31 @@ struct cvk_device : public _cl_device_id,
     bool supports_subgroups() const { return m_has_subgroups_support; }
 
     bool supports_non_uniform_decoration() const {
-        return m_properties.apiVersion >= VK_MAKE_VERSION(1, 2, 0) ||
-               is_vulkan_extension_enabled(
-                   VK_EXT_DESCRIPTOR_INDEXING_EXTENSION_NAME);
+        bool supported = m_properties.apiVersion >= VK_MAKE_VERSION(1, 2, 0) ||
+                         is_vulkan_extension_enabled(
+                             VK_EXT_DESCRIPTOR_INDEXING_EXTENSION_NAME);
+
+        // This is a hack to work-around a driver bug in mali device using the
+        // bifrost driver.
```

Comment:
```suggestion
```

This is at risk of becoming out of date. I suggest to delete the comment a replace it by something more generic.